utils/connectivity/conn_http.c

```diff
@@ -0,0 +1,236 @@
+    info->entropy = (mbedtls_entropy_context *)malloc(sizeof(mbedtls_entropy_context));
+    info->ctr_drbg = (mbedtls_ctr_drbg_context *)malloc(sizeof(mbedtls_ctr_drbg_context));
+    info->ssl_ctx = (mbedtls_ssl_context *)malloc(sizeof(mbedtls_ssl_context));
+    info->ssl_config = (mbedtls_ssl_config *)malloc(sizeof(mbedtls_ssl_config));
+    info->cacert = (mbedtls_x509_crt *)malloc(sizeof(mbedtls_x509_crt));
+
+    mbedtls_net_init(info->net_ctx);
+    mbedtls_ssl_init(info->ssl_ctx);
+    mbedtls_ssl_config_init(info->ssl_config);
+    mbedtls_x509_crt_init(info->cacert);
+    mbedtls_ctr_drbg_init(info->ctr_drbg);
+    mbedtls_entropy_init(info->entropy);
+
+    ret = mbedtls_ctr_drbg_seed(info->ctr_drbg, mbedtls_entropy_func, info->entropy, (const unsigned char *)seed_nonce,
+                                strlen(seed_nonce));
+    if (ret != 0) {
+      free(info->net_ctx);
+      free(info->ssl_ctx);
+      free(info->ssl_config);
+      free(info->ctr_drbg);
+      free(info->entropy);
+      free(info->cacert);
+      return RET_HTTP_INIT;
+    }
+
+    // TODO Change the PEM file path
+    ret = mbedtls_x509_crt_parse(info->cacert, (const unsigned char *)mbedtls_test_cas_pem, mbedtls_test_cas_pem_len);
+    if (ret < 0) {
+      printf("error: mbedtls_x509_crt_parse returned -0x%x\n\n", -ret);
+      return RET_HTTP_CERT;
+    }
+  }
+
+  ret = mbedtls_net_connect(info->net_ctx, host, port, MBEDTLS_NET_PROTO_TCP);
+  if (ret != 0) {
+    printf("error: mbedtls_net_connect returned %d\n\n", ret);
+    return RET_HTTP_CONNECT;
+  }
+
+  ret = mbedtls_ssl_config_defaults(info->ssl_config, MBEDTLS_SSL_IS_CLIENT, MBEDTLS_SSL_TRANSPORT_STREAM,
+                                    MBEDTLS_SSL_PRESET_DEFAULT);
+  if (ret != 0) {
+    printf("error: mbedtls_ssl_config_defaults returned %d\n\n", ret);
+    return RET_HTTP_SSL;
+  }
+
+  // TODO we may need to set the mode into MBEDTLS_SSL_VERIFY_REQUIRED
+  mbedtls_ssl_conf_authmode(info->ssl_config, MBEDTLS_SSL_VERIFY_OPTIONAL);
+
+  mbedtls_ssl_conf_rng(info->ssl_config, mbedtls_ctr_drbg_random, info->ctr_drbg);
+  mbedtls_ssl_conf_dbg(info->ssl_config, mbedtls_debug, stdout);
+
+  ret = mbedtls_ssl_setup(info->ssl_ctx, info->ssl_config);
+  if (ret != 0) {
+    printf("error: mbedtls_ssl_setup returned %d\n\n", ret);
+    return RET_HTTP_SSL;
+  }
+
+  ret = mbedtls_ssl_set_hostname(info->ssl_ctx, host);
+  if (ret != 0) {
+    printf("error: mbedtls_ssl_set_hostname returned %d\n\n", ret);
+    return RET_HTTP_SSL;
+  }
+
+  // Here is Blocking mode
+  mbedtls_ssl_set_bio(info->ssl_ctx, info->net_ctx, mbedtls_net_send, mbedtls_net_recv, mbedtls_net_recv_timeout);
+
+  ret = mbedtls_ssl_handshake(info->ssl_ctx);
+  while (ret != 0) {
+    if (ret != MBEDTLS_ERR_SSL_WANT_READ && ret != MBEDTLS_ERR_SSL_WANT_WRITE) {
+      printf("error: mbedtls_ssl_handshake returned -0x%x\n\n", -ret);
+      return RET_HTTP_SSL;
+    }
+  }
+
+#if 0
+  // Verify the server X.509 certificate
```

Comment:
Why did you leave dead code here?

Reply:
Pick the commit from [fix: Error in verifying the server X.509 certificate #72](https://github.com/DLTcollab/TA-endpoint/pull/72)?

Reply:
Have a clean codebase for collaboration.